src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/IAgentApi.cs

```diff
@@ -275,6 +275,24 @@ public interface IAgentApi
         /// </summary>
         /// <param name="callback">The callback to invoke to define the error group that an Exception belongs to.</param>
         void SetErrorGroupCallback(Func<IReadOnlyDictionary<string, object>, string> callback);
+
+        /// <summary> Sets the method that will be invoked to define the token count on completion.
+        ///
+        /// The callback takes the model name and input value, and returns an integer of the token count.
+        /// A value returned from the callback that is less than or equal to 0 will be ignored.
+        /// </summary>
+        /// <param name="callback">The callback to invoke to generate the token count based on the model and input..</param>
+        void SetLlmTokenCountingCallback(Func<string, string, int> callback);
+
+        /// <summary>
+        /// Creates an event with the customer feedback on the LLM interaction.
+        /// </summary>
+        /// <param name="traceId">Required. ID of the trace where the chat completion(s) related to the feedback occurred</param>
+        /// <param name="rating">Required. Rating provided by an end user. Must be string or int</param>
+        /// <param name="category">Optional. Category of the feedback as provided by the end user</param>
+        /// <param name="message">Optional. Freeform text feedback from an end user</param>
+        /// <param name="metadata">Optional. Set of key-value pairs to store any other desired data to submit with the feedback event</param>
+        //void RecordLlmFeedbackEvent(string traceId, object rating, string category = "", string message = "", IDictionary<string, object>? metadata = null)
```

Comment:
Do we still need this commented out code?

Reply:
We'll eventually implement feedback support, but not in this release. So I think it's fine as a placeholder for when we come back to it.

Reply:
We have implemented it, and there are passing integration tests for it, they just aren't being invoked via the interface. I think we can probably un-comment this out?